appIndicator.js

```diff
@@ -352,8 +352,6 @@ class AppIndicators_IconActor extends St.Icon {
     // collector.
     _cacheOrCreateIconByName(iconSize, iconName, themePath, callback) {
         let {scale_factor} = St.ThemeContext.get_for_stage(global.stage);
-        let maxSize = Convenience.getSettings().get_int('icon-size');
```

Comment:
Mmhmh, please don't add this in the first place, there's no point for uptream history to add and then remove a change in the same branch.

Reply:
Cleaned up all commits by squashing appropriate changes.
Now the exact history of how this has been developed is not visible anymore, but that is not useful anyhow.
Kept small fixes as these commits carry code explanations.